src/cosmosdb-preview/azext_cosmosdb_preview/custom.py

```diff
@@ -72,6 +77,212 @@ def _handle_exists_exception(cloud_error):
+                                    rule_name,
+                                    start_ip_address,
+                                    end_ip_address):
+
+    '''Creates an Azure Cosmos DB Mongo Cluster Firewall rule'''
+    
+    mongo_cluster_firewallRule = client.get_firewall_rule(resource_group_name, cluster_name, rule_name)
+
+    if start_ip_address is None:
+        start_ip_address = mongo_cluster_firewallRule.startIpAddress
+    
+    if end_ip_address is None:
+        firewall_rule_end_ip_address = mongo_cluster_firewallRule.endIpAddress
+    
+    firewall_rule =  FirewallRule(start_ip_address=start_ip_address, end_ip_address=end_ip_address)
+    
+    return client.begin_create_or_update_firewall_rule(resource_group_name, cluster_name, rule_name, firewall_rule)
+
+def cli_cosmosdb_mongocluster_firewall_rule_list(client, resource_group_name, cluster_name):
+
+    """List Azure CosmosDB Mongo Cluster Firewall Rule."""
+
+    return client.list_firewall_rules(resource_group_name, cluster_name)
+
+def cli_cosmosdb_mongocluster_firewall_rule_get(client, resource_group_name, cluster_name, rule_name):
+
+    """Gets Azure CosmosDB Mongo Cluster Firewall rule"""
+
+    return client.get_firewall_rule(resource_group_name, cluster_name, rule_name)
+
+def cli_cosmosdb_mongocluster_firewall_rule_delete(client, resource_group_name, cluster_name, rule_name):
+
+    """Delete Azure CosmosDB Mongo Cluster Firewall Rule"""
+
+    return client.begin_delete_firewall_rule(resource_group_name, cluster_name, rule_name)
+
+def cli_cosmosdb_mongocluster_create(client,
+                                    resource_group_name,
+                                    cluster_name,
+                                    administrator_login,
+                                    administrator_login_password,
+                                    location,
+                                    tags=None,
+                                    create_mode=CreateMode.DEFAULT.value,
+                                    restore_point_in_time_utc=None,
+                                    restore_source_resource_id=None,
+                                    server_version="5.0",
+                                    shard_node_sku=None,
+                                    shard_node_disk_size_gb=None, 
+                                    shard_node_ha=None,
+                                    shard_node_name=None,
+                                    shard_kind=NodeKind.SHARD.value,
+                                    shard_node_count=1):
+
+    '''Creates an Azure Cosmos DB Mongo Cluster '''
+    
+    if administrator_login is None:
+        raise InvalidArgumentValueError('Initial Mongo Cluster Admin user is required.')
+
+    if administrator_login_password is None :
+        raise InvalidArgumentValueError('Initial Mongo Cluster Admin Password is required.')
+
+    if restore_point_in_time_utc is not None and restore_source_resource_id is None:
```

Comment:
Use not(<exclusive_or_condition>) so that this is one if statement.

Reply:
fixed